CorsixTH/Lua/humanoid_actions/knock_door.lua

```diff
@@ -27,6 +27,11 @@ local KnockDoorAction = _G["KnockDoorAction"]
 --!param door (Object) Door to knock on.
 --!param direction (string) Direction of facing.
 function KnockDoorAction:KnockDoorAction(door, direction)
+  assert(class.is(door, Door), "expecting door object")
+  assert(direction == "north" or direction == "south" or
```

Comment:
maybe refactor the direction test out into a separate function? it seems to happen a lot.